tests/test_project.py

```diff
@@ -647,6 +648,80 @@ def test_reuse_info_of_copyright_xor_licensing(empty_directory):
     assert not bar_file_info.spdx_expressions
 
 
+def test_reuse_info_of_copyright_xor_licensing_with_license_dot(
+    empty_directory,
+):
+    """Test a corner case where partial REUSE information is defined inside of a
+    .license file (copyright xor licensing). Get the missing information from
+    the REUSE.toml.
+    """
+    (empty_directory / "REUSE.toml").write_text(
+        cleandoc(
+            """
+            version = 1
+
+            [[annotations]]
+            path = "*.py"
+            SPDX-FileCopyrightText = "2017 Jane Doe"
+            SPDX-License-Identifier = "CC0-1.0"
+
+            [[annotations]]
+            path = "*.py.license"
+            SPDX-FileCopyrightText = "2017 Jane Doe"
+            SPDX-License-Identifier = "MIT"
```

Comment:
This is a little strange to me, and seems incorrect. A `.license` file cannot/does not have licensing itself. The logic of REUSE is that these `.license` files are direct substitutes licensing-wise. This means that `foo.py.license` completely overrides the REUSE information inside of `foo.py`, and whenever `foo.py` is referenced anywhere (in REUSE.toml or in the code), we should really be looking at the contents of `foo.py.license` instead.

The second `[[annotations]]` table should probably be removed.

Reply:
The old behaviour would look up the information for the .license file path in the REUSE.toml instead of looking up the non .license file path. Because of that I thought i would add an explicit test that the .license file information in the REUSE.toml is ignored.

Reply:
I had another look at this. It took me a while to get it, but I think I understand. You're testing here that `.license` files are not valid targets inside of a `REUSE.toml` files, and should therefore be ignored.

I've moved this to a different test, to reduce the amount of stuff happening inside of one test. I've also simplified the initial test a little by reducing the amount of files, and by using `aggregate` instead of `closest`.

Thanks for your contribution!